secretstores/hashicorp/vault/vault_test.go

```diff
@@ -409,10 +409,49 @@ func getCertificate() []byte {
 }
 
 func TestGetFeatures(t *testing.T) {
-	s := NewHashiCorpVaultSecretStore(logger.NewLogger("test"))
-	// Yes, we are skipping initialization as feature retrieval doesn't depend on it.
-	t.Run("Vault supports MULTIPLE_KEY_VALUES_PER_SECRET", func(t *testing.T) {
+	initVaultWithVaultValueType := func(vaultValueType string) secretstores.SecretStore {
+		properties := map[string]string{
+			"vaultToken":     expectedTok,
+			"skipVerify":     "true",
+			"vaultValueType": vaultValueType,
+		}
+
+		m := secretstores.Metadata{
+			Base: metadata.Base{Properties: properties},
+		}
+
+		target := &vaultSecretStore{
+			client: nil,
+			logger: nil,
+		}
+
+		// This call will throw an error on Windows systems because of the of
```

Comment:
should we skip + warning when the runtime.GOOS == "windows" ?

Reply:
I really don't know how we should handle this. This particular comment is present many other tests in this file. I wonder how much effort should be done to make tests work for windows.

Is this just a matter of

```go
if err := target.Init(m) ; err != nil && runtime.GOOS != "windows" {
    t.Fatal(err)
}
```

Reply:
I agree with @mcandeia .

Just add a check such as 

```go
if runtime.GOOS != "windows" {
    t.Log("Skipping test on Windows")
    t.Skip()
}
```

(Skip, don't fatal :) )

Reply:
I will run this test on windows - this might be fixed for Go 1.18 --  and we require 1.19 anyway.

* https://github.com/golang/go/issues/16736
* https://github.com/rancher/system-agent/pull/84

Reply:
Things got interesting: we don't have that error on windows anymore apparently. But because errors were being bluntly ignored, the code was masking other errors. I am addressing these and improving the test itself.

I can address this in a new PR or bundle onto this one. What you folks prefer?

Reply:
https://github.com/dapr/components-contrib/pull/2331 Was merged. Resolving this thread.